brig/cmd/brig/commands/project_create.go

```diff
@@ -0,0 +1,537 @@
+				},
+			},
+			{
+				Name: "pullPolicy",
+				Prompt: &survey.Select{
+					Message: "Image Pull Policy",
+					Help:    "The image pull policy determines how often Kubernetes will try to refresh this image",
+					Options: []string{
+						"IfNotPresent",
+						"Always",
+						"Never",
+					},
+					Default: p.Worker.PullPolicy,
+				},
+			},
+		}, &p.Worker); err != nil {
+			return fmt.Errorf(abort, err)
+		}
+	}
+
+	advanced := false
+	if err := survey.AskOne(&survey.Confirm{
+		Message: "Configure Advanced Brigade Settings?",
+		Help:    "Set advanced options for Brigade",
+	}, &advanced, nil); err != nil {
+		return fmt.Errorf(abort, err)
+	} else if advanced {
+		if err := survey.Ask([]*survey.Question{
+			{
+				Name: "initGitSubmodules",
+				Prompt: &survey.Confirm{
+					Message: "Initialize Git Submodules",
+					Help:    "For repos that have submodules, initialize them on each clone. Not recommended on public repos.",
+					Default: p.InitGitSubmodules,
+				},
+			},
+			{
+				Name: "allowHostMounts",
+				Prompt: &survey.Confirm{
+					Message: "Allow host mounts",
+					Help:    "Allow host-mounted volumes for worker and jobs. Not recommended in multi-tenant clusters.",
+					Default: p.AllowHostMounts,
+				},
+			},
+			{
+				Name: "allowPrivilegedJobs",
+				Prompt: &survey.Confirm{
+					Message: "Allow privileged jobs",
+					Help:    "Allow jobs to mount the Docker socket or perform other privileged operations. Not recommended for multi-tenant clusters.",
+					Default: p.AllowPrivilegedJobs,
+				},
+			},
+			{
+				Name: "imagePullSecrets",
+				Prompt: &survey.Input{
+					Message: "Image Pull Secrets",
+					Help:    "Comma-separated list of image pull secret names that will be supplied to workers and jobs",
+					Default: p.ImagePullSecrets,
+				},
+			},
+			{
+				Name: "workerCommand",
+				Prompt: &survey.Input{
+					Message: "Worker Command",
+					Help:    "Override the worker's default command (yarn start). For debugging/expert use",
+					Default: p.WorkerCommand,
+				},
+			},
+			{
+				Name: "defaultScriptName",
+				Prompt: &survey.Input{
+					Message: "Default Script ConfigMap Name",
+					Help:    "It is possible to store a default script in a ConfigMap. Supply the name of that ConfigMap to use the script.",
+					Default: p.DefaultScriptName,
+				},
+			},
+		}, p); err != nil {
+			return fmt.Errorf(abort, err)
+		}
+		var fname string
+		err := survey.AskOne(&survey.Input{
+			Message: "Upload a Default brigade.js Script",
+			Help:    "The local path to a default brigade.js file that will be run if none exists in the repo. Overrides the ConfigMap script.",
+		}, &fname, loadFileValidator)
+		if err != nil {
+			return fmt.Errorf(abort, err)
+		}
+		if script := loadFileStr(fname); script != "" {
+			p.DefaultScript = loadFileStr(fname)
+		}
+	}
+
+	return nil
+}
+
+// loadFileValidator validates that a file exists and can be read.
+func loadFileValidator(val interface{}) error {
```

Comment:
I see why we have to do this, but do you have any ideas on if there are any cleaner ways to this approach? it feels a little awkward to accept an `interface{}` then immediately cast it into a string.

Reply:
I believe that to satisfy the Validator interface, we have to do `func (interface{}) error` as the signature.

Reply:
Yeah, I tried brainstorming if there were any Go hacks we could do to make this cleaner, but this looks as good as it gets. :)